docs/distribution/fallback-launcher-release-infrastructure.md

```diff
@@ -0,0 +1,143 @@
+        "enso-launcher-0.1.0-test-linux-amd64.tar.gz"
+      ]
+    }
+  ]
+}
+```
+
+The launcher package should be accessible under the URL
+`https://launcherfallback.release.enso.org/launcher/0.1.0-test/enso-launcher-0.1.0-test-linux-amd64.tar.gz`.
+
+The fallback infrastructure should be able to provide all available launcher
+releases. If for some reason that were infeasible, the absolute minimum is to
+provide at least such a set of releases that will allow every launcher version
+since its release to do a
+[step-by-step upgrade](launcher.md#step-by-step-upgrade) to the latest available
+version.
+
+### Requirements Inherited from Current Launcher Upgrade Scheme
+
+The fallback releases must also adhere to the release scheme as it was defined
+when the first launcher version was released. That brings the following
+requirements:
+
+- The version is a semantic versioning string.
+- The tag has format `enso-<version>`.
+- The release contains an asset called `launcher-manifest.yaml` which contains
+  at least one key, `minimum-version-for-upgrade` which is a semantic versioning
+  string.
+- The release contains the following packages:
+  - `enso-launcher-<version>-linux-amd64.tar.gz`
+  - `enso-launcher-<version>-macos-amd64.tar.gz`
+  - `enso-launcher-<version>-windows-amd64.zip`
+
+## Current Fallback Infrastructure Implementation
+
+Currently the fallback mechanism is disabled, as it should only be enabled in
+case the primary release provider becomes permanently unavailable (or at least
+in case the unavailability is not temporary and is known to span a long enough
+time-frame that a decision is made that the fallback is necessary).
+
+However, to be sure that we can easily enable it, the infrastructure is already
+in a working state. It is built on top of an AWS S3 bucket which provides the
+files explained in the
+[specification above](#fallback-infrastructure-specification). Each release
+uploaded to GitHub Releases is automatically uploaded to this S3 bucket so it
+also acts as a backup.
+
+If it is necessary, all that needs to be done to enable the fallback is to set
+the `enabled` property in the manifest to `true`.
+
+### Updating the Release List
+
+When a new release is built on the CI, its packages are uploaded to the S3
+bucket, but also the `release-list.json` has to be updated. This is handled by
+downloading the file, appending to it and re-uploading it. This mechanism is not
+safe from race conditions if multiple release jobs were running at the same
+time, so we require that release jobs are invoked serially (and since new
+releases are not added extremely frequently, this should not be an issue
+currently). If this ever becomes a problem, a more complex solution can be
+developed which synchronizes access to the release list.
+
+### Marking the Release As Broken
+
+When a release is marked as broken on GitHub (by uploading a file called
+`broken` to the assets), a GitHub Action is triggered which uploads the same
```

Comment:
Should specify that the file should be empty.

Reply:
Actually it seems that empty files cannot be uploaded to GitHub releases.

I get: 
![image](https://user-images.githubusercontent.com/1436948/92935751-83d84480-f449-11ea-9ce9-d0007b54823d.png)

Isn't it fine for the file content to be unspecified since we only check if it exists?